README.md

```diff
@@ -4,47 +4,109 @@
+
+The name uftrace stems from the combination of user and ftrace:
+
+It can record data from:
+- User space functions by dynamically patching functions at runtime, or with code
+  compiled with with `-pg`, `-fpatchable-function-entry`, `-mfentry`, `-mnop-mcount`,
+  `-finstrument-functions`, or using clang with `-fxray-instrument`.
+- Library functions (through PLT hooking)
+- Kernel functions (using the ftrace framework in Linux kernel)
+- Kernel trace events affecting the program, e.g. scheduling events (which affect the
+  execution timing of the program) and records nanosecond-exact timestamps
+  (using perf_event and systemtap SDT)
+
+With the recorded data, it:
+- Can show colored and nested function call graphs
+- Can show arguments and return values symbolically using libc function prototypes
+  and DWARF debug information.
+- Can apply filters like minimal function call duration
+- Can extract metadata from traces (e.g. system information on which the trace was taken)
+- Can generate a symbol table of the traced program and library functions
+- When the execution of multiple programs was recorded,
+  it can generate a task relationship tree (parent and children) from the trace.
+
+It supports many commands and filters such as filtering by function call
+duration for analysis of program execution and performance.
 
 ![uftrace-live-demo](doc/uftrace-live-demo.gif)
 
  * Homepage: https://github.com/namhyung/uftrace
  * Tutorial: https://github.com/namhyung/uftrace/wiki/Tutorial
  * Chat: https://gitter.im/uftrace/uftrace
  * Mailing list: [uftrace@googlegroups.com](https://groups.google.com/forum/#!forum/uftrace)
+ * Lightning talk: https://youtu.be/LNav5qvyK7I
 
 
 Features
 ========
 
-It traces each function in the executable and shows time duration.  It
-can also trace external library calls - but usually entry and exit are
-supported.  Optionally it's possible to trace other (nested) external
-library calls and/or internal function calls in the library call.
-
-It can show detailed execution flow at function level, and report which
-function has the highest overhead.  And it also shows various information
-related the execution environment.
+uftrace traces each function in the executable and shows time durations.
+
+Usually, for this to be possible, the program needs to be compiled with
+`-pg` or `-fpatchable-function-entry=5`, but with full dynamic tracing
+(`-P.`|`--patch=.`), uftrace works on all executables (as long they are
+not stripped, or symbol information is available from a separate file).
+
+uftrace is able to perform nested library call tracing even on stripped
+executables with no symbol information using (`-l`|`--nest-libcall`)
+by hooking into the procedure linkage tables (PLTs) of shared libraries.
+The depth can be limited using `-D<num>`, where 1 is flat call tracing.
+
+Using (`-a`|`--auto-args`), uftrace automatically records arguments and
+return values of known functions. Without extra debug information, this
```

Comment:
In addition, `-a` implies `--srcline` so it records the source line location info and it can be shown in `uftrace replay --srcline` and in `uftrace tui`, users can directly open their editor with the exact source location as shown in https://uftrace.github.io/slide/#120.

Reply:
added.